go-controller/pkg/ovn/gateway.go

```diff
@@ -278,6 +278,19 @@ func (gw *GatewayManager) GatewayInit(
 	if gw.netInfo.GetNetworkName() == types.DefaultNetworkName {
 		logicalRouterOptions["snat-ct-zone"] = "0"
 	}
+	if gw.netInfo.TopologyType() == types.Layer2Topology {
+		// When multiple networks are set of the same logical-router-port
+		// the networks get lexicographically sorted; thus there is no
+		// ordering or telling on which IP will be chosen as the router-ip
+		// when it comes to SNATing traffic after load balancing.
+		// Hence for Layer2 UDPNs let's set the snat-ip explicitly to the
+		// joinsubnetIP
+		joinIPDualStack := make([]string, len(gwLRPIfAddrs))
```

Comment:
I don't follow this: isn't the subnet GW IP also in this list for the L2 topology ? 
https://github.com/ovn-kubernetes/ovn-kubernetes/blob/f6fd10db3e3ca9db556ec69d898df0fcb5794bbf/go-controller/pkg/ovn/secondary_layer2_network_controller.go#L728

Reply:
ah. You're removing it further down :facepalm:

Reply:
Still, for this to make sense (to me) we'd need to rename the attributes: this no longer is the `gwLRPIfAddrs`, but the joinIPs. 

It becomes the `gwLRPIfAddrs` once you append the subnet GW further below.

Reply:
yea I have been battling this confusion as well, haha I fully agree with you here that a rename of the variables is necessary here.

Reply:
this is done